code/modules/antagonists/heretic/heretic_knowledge.dm

```diff
@@ -29,6 +29,10 @@
 	var/list/required_atoms
 	/// Paired with above. If set, the resulting spawned atoms upon ritual completion.
 	var/list/result_atoms = list()
+	/// If set this allows the heretic to add additional atoms to get different results than when only using required atoms
+	var/list/optional_atoms = list()
+	/// If set this makes the optional atoms give a different result
+	var/list/optional_result_atoms = list()
```

Comment:
something about this system smells wrong, could you explain how it's meant to work/why it exists?

Reply:
I'll tackle this in a separate pr.